src/services/PaymentSchedulerService.js

```diff
@@ -131,7 +131,7 @@ async function processPayment (workPeriodPayment) {
     await postEvent(config.TAAS_WORK_PERIOD_PAYMENT_UPDATE_TOPIC, updated.toJSON(), { oldValue })
 
     if (paymentScheduler) {
-      await paymentScheduler.update({ step: PaymentSchedulerStatus.CLOSE_CHALLENGE, userId: paymentScheduler.userId, status: 'failed' })
+      await paymentScheduler.update({ step: _.get(err, 'step', PaymentSchedulerStatus.CLOSE_CHALLENGE), userId: paymentScheduler.userId, status: 'failed' })
```

Comment:
@xxcxy can we set `undefined` or `null` in case we didn't get step from error? I'm a bit afraid to always fall back to `close-challenge` as it might hide issues in our logic. What if some error happened in some other step, but somehow we didn't pass the correct `step` and it says `close-challenge` but actually it's different.

Reply:
Since the previous definition of this table is only used to restore the task of abnormal exit, there is no strict consistency design for the data of this table in other situations. If you want to make it undefined or null, we need to change this field to allow null.